###Define the function of distance to last scattering:

In [2]:

def H_over_h0(z, ombh2, omh2, w, h0):
    om = (omh2 + ombh2) / h0**2
    om_rad = 5.46 * 10**-5
    return np.sqrt((om) * (1 + z)**3 + om_rad** (1 + z)**4 + (1 - om) * (1 + z)**(3. * (1. + w)))


def Da(z, ombh2, omh2, w, h0):
    '''comoving distance'''
    from scipy.integrate import quad
    def integrand(z, ombh2, omh2, w, h0):
        return (3000 / h0) / (H_over_h0(z, ombh2, omh2, w, h0))
    return quad(integrand, 0, 1090, args=(ombh2, omh2, w, h0), epsrel=1.49e-11)[0]

####Derivative are taken with scipy

In [3]:
from scipy.misc import derivative

####Trick to do partial derivatives.

In [4]:
step = 0.05
def partial_derivative(func, var=0, point=[]):
    args = point[:]
    def wraps(x):
        args[var] = x
        return func(*args)
    return derivative(wraps, point[var], dx = point[var]*step,order=7 )

In [5]:
args = [1100,0.022068,0.12029,-1,0.67]
sigma_args = [0,0.1,0.4,9,3.8]

## Results

In [6]:
partial_derivative(Da, 1, args)

-39674.271476901733

In [7]:
partial_derivative(Da, 2, args)

-39674.274030200788

In [8]:
partial_derivative(Da, 3, args)

-1178.93597043969

In [9]:
partial_derivative(Da, 4, args)

-4016.8171697005159

Now $$\Delta D_{LSS} = -39674 [\Delta\Omega_{b}h^{2}/\Omega_{b}h^{2} ]\Omega_{b}h^{2} -39674 [\Delta\Omega_{c}h^{2}/\Omega_{c}h^{2} ]\Omega_{c}h^{2}-1178 [\Delta w/w ]w-4016[\Delta H_{0}/H_{0} ]H_{0} $$

In [10]:
partial_derivative(Da, 1, args)*sigma_args[1]*args[1]

-87.553182295226748

In [11]:
partial_derivative(Da, 2, args)*sigma_args[2]*args[2]

-1908.9673692371412

In [12]:
partial_derivative(Da, 3, args)*sigma_args[3]*args[3]

10610.42373395721

In [13]:
partial_derivative(Da, 4, args)*sigma_args[4]*args[4]

-10226.816514057515

So: $$ \Delta D_{LSS} = -87 \times  (\frac{\sigma(\Omega_{b}h^{2})}{0.1 \%})\times0.022 -1908 \times  ( \frac{\sigma(\Omega_{c}h^{2})}{0.4 \%})\times0.12 +10610 \times  (\frac{\sigma(w)}{9 \%})\times -1 - 10226 \times  (\frac{\sigma(H_{0})}{3.8 \%})\times 0.67 $$